daemon/logger/ring.go

```diff
@@ -111,13 +114,15 @@ func (r *RingLogger) Close() error {
 			logErr = true
 		}
 	}
+	r.wg.Wait()
```

Comment:
Maybe this should go after setClosed?

Reply:
It works either ways.. buffer.Close() has the right semantics to not have a race condition. I will move it up if that is more preferred.

Reply:
My thought here is to not have two things attempting to log at the same time.

Reply:
Sounds good.. I moved it after the `setClosed()` call.